server/activityItem/queries.ts

```diff
@@ -0,0 +1,176 @@
+import * as types from 'types';
+
+import { Pub, ActivityItem, Collection, Member, Community, CollectionPub } from 'server/models';
+
+const createActivityItem = (ai: types.InsertableActivityItem) => ActivityItem.create(ai);
+
+const getDiffsForPayload = <
+	Entry extends Record<string, any>,
+	EntryKey extends keyof Entry,
+	Diffs = Partial<{ [Key in EntryKey]: types.Diff<Entry[Key]> }>
+>(
+	newEntry: Entry,
+	oldEntry: Entry,
+	keys: EntryKey[],
+): Diffs => {
+	return keys.reduce(
+		(memo: Diffs, key: EntryKey) =>
+			oldEntry[key] === newEntry[key]
+				? memo
+				: {
+						...memo,
+						[key]: {
+							from: oldEntry[key],
+							to: newEntry[key],
+						},
+				  },
+		{} as Diffs,
+	);
+};
+
+const getChangeFlagsForPayload = <
+	Entry extends Record<string, any>,
+	EntryKey extends keyof Entry,
+	Flags = Partial<{ [Key in EntryKey]: true }>
+>(
+	newEntry: Entry,
+	oldEntry: Entry,
+	keys: EntryKey[],
+): Flags => {
+	return keys.reduce(
+		(memo: Flags, key: EntryKey) =>
+			oldEntry[key] === newEntry[key] ? memo : { ...memo, [key]: true },
+		{} as Flags,
+	);
+};
+
+export const createPubActivityItem = async (
+	kind: 'pub-created' | 'pub-updated' | 'pub-removed' | 'pub-released',
+	userId: string,
+	oldPub: types.Pub,
+	communityId: string,
+	releaseId: string,
+	pubId: string,
+) => {
+	const pub: types.Pub = Pub.findOne({ where: pubId });
```

Comment:
Need an `await` here (and in a few other equivalent spots)

Reply:
Also, food for thought...it may be desirable to avoid making this call in the `pub-created` and `pub-removed` cases, since nothing will actually be diffed here.